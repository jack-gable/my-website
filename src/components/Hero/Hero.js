"use client";
import React from "react";
import peep from "../../../public/assests/img/peep-46.svg";
import Image from "next/image";
import styles from "./Hero.module.css";
import Link from "next/link";
import { FiArrowRight as Arrow } from "react-icons/fi";
import TypewriterEffect from "../Typewriter";

const words = [
	{
		text: "Hi",
	},
	{
		text: "I'm",
	},
	{
		text: "Jack",
	},
	{
		text: "Gable",
	},
];

function Hero() {
	return (
		<div className={styles.wrapper}>
			<div>
				<div className={styles.flexWrapper}>
					<div className={styles.container}>
						<TypewriterEffect words={words} />
						<p style={{ margin: 0 }}>
							I&apos;m a developer and tech enthusiast based in
							California.
						</p>
						<div
							className={`${styles.linksWrapper} ${styles.paddingTop}`}
						>
							<Link href="/projects" className={styles.link}>
								view my projects
								<span className={styles.arrow}>
									<Arrow strokeWidth={3} />
								</span>
							</Link>
							<Link href="/contact" className={styles.link}>
								Contact Me
								<span className={styles.arrow}>
									<Arrow strokeWidth={3} />
								</span>
							</Link>
						</div>
					</div>
					<Image
						className={styles.peep}
						src={peep}
						alt=""
						height={300}
						width={200}
						priority={true}
						data-testid="hero-peep"
					/>
				</div>
			</div>
		</div>
	);
}

export default Hero;
