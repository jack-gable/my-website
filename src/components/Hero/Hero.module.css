.wrapper {
	width: 100%;
	max-width: 750px;
	padding: 12rem 0;
	display: grid;
	place-content: center;
	margin-right: auto;
	margin-left: auto;
}

.title {
	font-size: 3.5rem;
}

.linksWrapper {
	width: 60%;
	display: flex;
	justify-content: space-between;
}

.arrow {
	display: inline-block;
	color: var(--color-primary);
	transition: transform 200ms;
}

.link {
	font-weight: 600;
	text-transform: capitalize;
	display: flex;
	align-items: center;
	gap: 4px;
}

.link:hover {
	text-underline-offset: 0.4em;
}

.link:hover .arrow {
	transform: translateX(6px);
}

.flexWrapper {
	display: flex;
	justify-content: center;
	align-items: center;
}

.peep {
	filter: drop-shadow(2px 4px 8px var(--color-decorative-500));
}

.paddingTop {
	padding-top: 24px;
}

@media (max-width: 35rem) {
	.wrapper {
		padding: 6rem 0;
	}

	.flexWrapper {
		flex-direction: column;
	}

	.container {
		padding-left: 1.2rem;
		padding-right: 1.2rem;
	}

	.link {
		font-size: 0.8rem;
	}

	.linksWrapper {
		width: 100%;
		justify-content: revert;
		gap: 2rem;
	}
}
