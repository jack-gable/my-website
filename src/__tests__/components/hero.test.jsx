import React from "react";
import Hero from "@/components/Hero";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import "intersection-observer";

describe("<Hero />", () => {
	it("renders the <Hero /> component with text", () => {
		const { container, getByText, getByTestId } = render(<Hero />);

		expect(
			getByText(
				"I'm a developer and tech enthusiast based in California."
			)
		).toBeInTheDocument();
		expect(getByText("view my projects")).toBeInTheDocument();
		expect(getByText("Contact Me")).toBeInTheDocument();
		expect(getByTestId("hero-peep")).toBeInTheDocument();
		expect(container.firstChild).toMatchSnapshot();
	});
});
