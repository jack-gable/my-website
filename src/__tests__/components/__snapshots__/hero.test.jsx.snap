// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> renders the <Hero /> component with text 1`] = `
<div
  class="wrapper"
>
  <div>
    <div
      class="flexWrapper"
    >
      <div
        class="container"
      >
        <div
          class="wordWrapper"
        >
          <div
            class="inline"
          >
            <div
              class="inlineBlock"
            >
              <span
                class="word"
              >
                H
              </span>
              <span
                class="word"
              >
                i
              </span>
               
            </div>
            <div
              class="inlineBlock"
            >
              <span
                class="word"
              >
                I
              </span>
              <span
                class="word"
              >
                '
              </span>
              <span
                class="word"
              >
                m
              </span>
               
            </div>
            <div
              class="inlineBlock"
            >
              <span
                class="word"
              >
                J
              </span>
              <span
                class="word"
              >
                a
              </span>
              <span
                class="word"
              >
                c
              </span>
              <span
                class="word"
              >
                k
              </span>
               
            </div>
            <div
              class="inlineBlock"
            >
              <span
                class="word"
              >
                G
              </span>
              <span
                class="word"
              >
                a
              </span>
              <span
                class="word"
              >
                b
              </span>
              <span
                class="word"
              >
                l
              </span>
              <span
                class="word"
              >
                e
              </span>
               
            </div>
          </div>
          <span
            class="cursor"
            style="opacity: 0;"
          />
        </div>
        <p
          style="margin: 0px;"
        >
          I'm a developer and tech enthusiast based in California.
        </p>
        <div
          class="linksWrapper paddingTop"
        >
          <a
            class="link"
            href="/projects"
          >
            view my projects
            <span
              class="arrow"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="5"
                  x2="19"
                  y1="12"
                  y2="12"
                />
                <polyline
                  points="12 5 19 12 12 19"
                />
              </svg>
            </span>
          </a>
          <a
            class="link"
            href="/contact"
          >
            Contact Me
            <span
              class="arrow"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="5"
                  x2="19"
                  y1="12"
                  y2="12"
                />
                <polyline
                  points="12 5 19 12 12 19"
                />
              </svg>
            </span>
          </a>
        </div>
      </div>
      <img
        alt=""
        class="peep"
        data-nimg="1"
        data-testid="hero-peep"
        decoding="async"
        fetchpriority="high"
        height="300"
        src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
    </div>
  </div>
</div>
`;
